src/App.tsx

```diff
@@ -15,36 +17,85 @@
   'Garlic',
 ];
 
+const SORT_BY_ALPH: number = 1;
+const SORT_BY_LENGTH: number = 2;
+
+const sortGoods = (goods: string[], key: number): string[] => {
+  switch (key) {
+    case SORT_BY_ALPH:
+      return goods.sort((a, b) => a.localeCompare(b));
+    case SORT_BY_LENGTH:
+      return goods.sort((a, b) => a.length - b.length);
+    default:
+      return goods;
+  }
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(0);
+  const [toReverse, setToReverse] = useState(false);
+
+  const goods = sortGoods(goodsFromServer.slice(), sortBy);
+
+  if (toReverse) {
+    goods.reverse();
+  }
+
+  const handleReverse = () => {
+    setSortBy(0);
+    setToReverse(false);
+  };
```

Comment:
[CODE STYLE] - Function's name should start with a verb. It's better to name your function `handleReset` instead of `handleReverse` to make it clear what the function does. Also, you don't need to set `sortBy` to 0 and `toReverse` to false here. You can do it directly in the `onClick` handler of the Reset button.